llvm/include/llvm/IR/IntrinsicsNVVM.td

```diff
@@ -5043,6 +5043,16 @@ def int_nvvm_prefetch_global_L2_evict_last: Intrinsic<[], [llvm_global_ptr_ty],
 def int_nvvm_prefetchu_L1 : Intrinsic<[], [llvm_ptr_ty],
   [IntrArgMemOnly, ReadOnly<ArgIndex<0>>, NoCapture<ArgIndex<0>>]>;
 
+def int_nvvm_applypriority_global_L2_evict_normal
+  : Intrinsic<[], [llvm_global_ptr_ty, llvm_i64_ty],
+              [IntrArgMemOnly, ReadOnly<ArgIndex<0>>, NoCapture<ArgIndex<0>>,
+               ImmArg<ArgIndex<1>>]>;
+
+def int_nvvm_applypriority_L2_evict_normal
```

Comment:
Perhaps it would be cleaner to make this a single overloaded intrinsic with ISel based on the type of the pointer arg?

Reply:
Overloaded intrinsic will create instances for pointers in all AS. AFAICT, these only apply to global pointers and their generic counterparts. Naming them explicitly is appropriate here, IMO.